backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/impl/PaySelectionBL.java

```diff
@@ -563,4 +564,29 @@ public void updatePaySelectionTotalAmt(@NonNull final PaySelectionId paySelectio
 	{
 		paySelectionDAO.updatePaySelectionTotalAmt(paySelectionId);
 	}
+
+	@Override
+	public PaySelectionLineType extractType(final I_C_PaySelectionLine line)
+	{
+		final PaymentId originalPaymentId = PaymentId.ofRepoIdOrNull(line.getOriginal_Payment_ID());
+		final InvoiceId invoiceId = InvoiceId.ofRepoIdOrNull(line.getC_Invoice_ID());
+		if (originalPaymentId != null)
+		{
+			return PaySelectionLineType.Refund;
+		}
+		else if (invoiceId != null)
+		{
+			return PaySelectionLineType.Invoice;
+		}
+		else
+		{
+			throw new AdempiereException("Unsupported type!");
```

Comment:
pls provide more context. at least we need to know which is the lineId that failed.

Reply:
(think to the support team)